tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/ReplaceFeedsInNuGetConfig.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    /// <summary>
+    /// Replaces feeds in a NuGet.Config file given a mapping
+    /// of old feeds to new feeds.
+    /// </summary>
+    public class ReplaceFeedsInNuGetConfig : Task
+    {
+        /// <summary>
+        /// The NuGet.Config file in which to replace feeds.
+        /// </summary>
+        [Required]
+        public string InputFile { get; set; }
+
+        /// <summary>
+        /// An item group of feeds to update.  Each item must have metadata NewFeed
+        /// that specifies the replacement feed.
+        /// </summary>
+        [Required]
+        public ITaskItem[] FeedMapping { get; set; }
```

Comment:
A way I like to abbreviate this and give it a little more structure is:

```C#
/// <summary>
/// An item group of feeds to update.
/// %(Identity): The feed URL to find in the NuGet.Config.
/// %(NewFeed): The feed URL to replace %(Identity) with.
/// </summary>
```

(In this case I think it's ok to leave as is, since it's easy to find usage with a global search in the repo for `FeedMapping` anyway.)